pkg/controllers/collaset/synccontrol/update.go

```diff
@@ -207,7 +213,12 @@ func decidePodToUpdateByLabel(_ *appsv1alpha1.CollaSet, podInfos []*PodUpdateInf
 	return podToUpdate
 }
 
-func decidePodToUpdateByPartition(cls *appsv1alpha1.CollaSet, podInfos []*PodUpdateInfo) (podToUpdate []*PodUpdateInfo) {
+func decidePodToUpdateByPartition(
```

Comment:
I think we should treat the NotCreated Pod as placeholder. These Pod placeholders should participate into order operation. Then it is able to get rid of `decideNeedUpdateButNotCreate`.

Reply:
fixed by constructing placeholder podwrappers for ownedIDs